src/query/interpreter.hpp

```diff
@@ -290,24 +298,35 @@ class Interpreter final {
 
   void BeginTransaction();
 
+  /*
+  Returns transaction id or empty if the db_accessor is not initialized.
+  */
+  std::optional<uint64_t> GetTransactionId() const;
+
   void CommitTransaction();
 
   void RollbackTransaction();
 
   void SetNextTransactionIsolationLevel(storage::IsolationLevel isolation_level);
   void SetSessionIsolationLevel(storage::IsolationLevel isolation_level);
 
+  std::vector<TypedValue> GetQueries();
+
   /**
    * Abort the current multicommand transaction.
    */
   void Abort();
 
+  std::atomic<TransactionStatus> transaction_status_{TransactionStatus::NO_TRANSACTION};
+
  private:
   struct QueryExecution {
     std::optional<PreparedQuery> prepared_query;
     utils::MonotonicBufferResource execution_memory{kExecutionMemoryBlockSize};
     utils::ResourceWithOutOfMemoryException execution_memory_with_exception{&execution_memory};
 
+    std::string query;
```

Comment:
Maybe rename this as active query or something like that. Or last query, I think that is better naming.